src/main/java/segelzwerg/sporttooolbox/IUnits/Time.java

```diff
@@ -2,12 +2,31 @@
 
 public class Time {
     private final int hour;
+    private final int minutes;
+    private final int seconds;
 
     public Time(int hour) {
+        this(hour, 0, 0);
+    }
+
+    public Time(int hour, int minutes) {
+        this(hour, minutes, 0);
+    }
+
+    public Time(int hour, int minutes, int seconds) {
+        if (hour < 0 || minutes < 0 || seconds < 0) {
+            throw new IllegalArgumentException("Times must not be negative");
+        }
         this.hour = hour;
+        this.minutes = minutes;
+        this.seconds = seconds;
     }
 
     public Speed computeSpeed(float kilometer, float meter) {
-        return new Speed((kilometer + meter / 1000) / hour);
+        return new Speed((kilometer + meter / 1000) / inHours());
+    }
+
+    protected float inHours() {
+        return hour + (minutes / 60f) + (seconds / 3_600f);
     }
```

Comment:
I think it would be better when this method is private.
What I meant with overflow ist this:
`new Time(2, 62, 62) == new Distance(3, 3, 2)`
I hope this makes it clearer and sorry what not being precise on that.

Reply:
Making this method private would prevent direct testing, so I'd rather make it package-private. If you really want to limit visibility as much as possible, I can make it private and test via computeSpeed.

I'm still not entirely clear on the overflow, I think. Should I add equals() and hashCode()? Because the following are equivalent now (even if the fields are not the same):

    Time a = new Time(2, 62, 62)
    Time b = new Time(3, 3, 2)

Reply:
We don't equal and has the are created by lombok with the Annotation `@EqualsAndHashCode` before the declaration of the Distance Class